src/emit/substrate/mod.rs

```diff
@@ -219,7 +219,7 @@ impl SubstrateTarget {
 
         // after copying stratch, first thing to do is abort value transfers if constructors not payable
         if abort_value_transfers {
-            abort_if_value_transfer(self, binary, function, ns);
+            abort_if_value_transfer(self, binary, function, ns, "constructor");
```

Comment:
I doubt this is correct. `public_function_prelude` is used during emitting the `deploy` (for constructors) and the `call` (for messages) function. The perfect solution would be to have the function name here. But for me for now it doesn't have to be perfect, just not wrong. Can you please just distinguish between "deploy" and "call" here  (can be added as a &'static str argument to `fn public_function_prelude`).